Error : The path "/tmp/migration.sql" not writeable!

running 
```
bin/console doctrine:migration:migrate --write-sql=/tmp/migration.sql --dry-run -vvv
```

generating error :
```
[ERROR] The path "/tmp/migration.sql" not writeable!
```

using _is_writable_ in the command return false if the file does not exist.
is it intentional to force the file to exist?

The workaround is to perform a touch on the file before run the script.

--- a/lib/Doctrine/Migrations/Tools/Console/Command/ExecuteCommand.php
+++ b/lib/Doctrine/Migrations/Tools/Console/Command/ExecuteCommand.php
@@ -12,8 +12,10 @@ use Symfony\Component\Console\Input\InputOption;
 use Symfony\Component\Console\Output\OutputInterface;
 
 use function array_map;
+use function dirname;
 use function getcwd;
 use function implode;
+use function is_dir;
 use function is_string;
 use function is_writable;
 use function sprintf;
@@ -130,7 +132,8 @@ EOT
             : Direction::UP;
 
         $path = $input->getOption('write-sql') ?? getcwd();
-        if (is_string($path) && ! is_writable($path)) {
+
+        if (is_string($path) && ! $this->isPathWritable($path)) {
             $this->io->error(sprintf('The path "%s" not writeable!', $path));
 
             return 1;
@@ -160,5 +163,10 @@ EOT
         $this->io->newLine();
 
         return 0;
+    }
+
+    private function isPathWritable(string $path): bool
+    {
+        return is_writable($path) || is_dir($path) || is_writable(dirname($path));
     }
 }

--- a/lib/Doctrine/Migrations/Tools/Console/Command/MigrateCommand.php
+++ b/lib/Doctrine/Migrations/Tools/Console/Command/MigrateCommand.php
@@ -15,8 +15,10 @@ use Symfony\Component\Console\Input\InputOption;
 use Symfony\Component\Console\Output\OutputInterface;
 
 use function count;
+use function dirname;
 use function getcwd;
 use function in_array;
+use function is_dir;
 use function is_string;
 use function is_writable;
 use function sprintf;
@@ -143,7 +145,8 @@ EOT
         $versionAlias     = $input->getArgument('version');
 
         $path = $input->getOption('write-sql') ?? getcwd();
-        if (is_string($path) && ! is_writable($path)) {
+
+        if (is_string($path) && ! $this->isPathWritable($path)) {
             $this->io->error(sprintf('The path "%s" not writeable!', $path));
 
             return 1;
@@ -278,5 +281,10 @@ EOT
         }
 
         return 0;
+    }
+
+    private function isPathWritable(string $path): bool
+    {
+        return is_writable($path) || is_dir($path) || is_writable(dirname($path));
     }
 }

--- a/tests/Doctrine/Migrations/Tests/Tools/Console/Command/ExecuteCommandTest.php
+++ b/tests/Doctrine/Migrations/Tests/Tools/Console/Command/ExecuteCommandTest.php
@@ -94,10 +94,12 @@ class ExecuteCommandTest extends MigrationTestCase
             [true, false, null],
             [true, null, getcwd()],
             [true,  __DIR__ . '/_files', __DIR__ . '/_files'],
+            [true,  __DIR__ . '/_files/run.sql', __DIR__ . '/_files/run.sql'],
 
             [false, false, null],
             [false, null, getcwd()],
             [false,  __DIR__ . '/_files', __DIR__ . '/_files'],
+            [true,  __DIR__ . '/_files/run.sql', __DIR__ . '/_files/run.sql'],
         ];
     }
 

